File: 6 kyu/Replace With Alphabet Position/solution.rb
def alphabet_position(text)
  text.gsub(/[^a-z]/i, '').chars.map{ |c| c.downcase.ord - 96 }.join(' ')
end

________________________________________________
def alphabet_position(text)
  # Delete everything but letters from the string
  only_letters = text.delete("^a-zA-Z")
  
  # Make every letter in the new string lowercase
  lower_case = only_letters.downcase
  
  # Convert each letter to byte position
  # Note: byte positions are sequential - subtract 96 from
  # each value and you get their position in the alphabet
  byte_value = lower_case.bytes
  
  # Produce new array using .map with alphabet position correct
  mapped_text = byte_value.map { |ltr| ltr - 96 }
  
  # Return final answer
  final_answer = mapped_text.join(' ')
end

________________________________________________
def alphabet_position(text)
  text.upcase.chars.select { |c| ("A".."Z").include?(c) } .map { |c| c.ord-64 } .join(" ")
end

________________________________________________
def alphabet_position(text)
  result = ''
  text.split('').each do |c|
    case c.downcase
      when 'a'
        result += '1 '
      when 'b'
        result += '2 '
      when 'c'
        result += '3 '
      when 'd'
        result += '4 '
      when 'e'
        result += '5 '
      when 'f'
        result += '6 '
      when 'g'
        result += '7 '
      when 'h'
        result += '8 '
      when 'i'
        result += '9 '
      when 'j'
        result += '10 '
      when 'k'
        result += '11 '
      when 'l'
        result += '12 '
      when 'm'
        result += '13 '
      when 'n'
        result += '14 '
      when 'o'
        result += '15 '
      when 'p'
        result += '16 '
      when 'q'
        result += '17 '
      when 'r'
        result += '18 '
      when 's'
        result += '19 '
      when 't'
        result += '20 '
      when 'u'
        result += '21 '
      when 'v'
        result += '22 '
      when 'w'
        result += '23 '
      when 'x'
        result += '24 '
      when 'y'
        result += '25 '
      when 'z'
        result += '26 '
      else
        result = result
    end
  end
  result[0..-2]
end

________________________________________________
def alphabet_position(text)
  alphabet = ("a".."z").to_a
  text = text.downcase.chars
  positions = []
  text.each do |char|
    if alphabet.include? char
      positions << alphabet.find_index(char).to_i + 1
    end
  end
  
  positions.join(" ")
end

File: 6 kyu/Replace With Alphabet Position/solution.py
def alphabet_position(text):
    res = []
    for c in text.lower():
        if ord(c) < ord('a') or ord(c) > ord('z'):
            continue
        res.append(str(ord(c) - ord('a') + 1))
    return ' '.join(res)
  
_______________________________
def alphabet_position(text):
    return ' '.join(str(ord(c) - 96) for c in text.lower() if c.isalpha())
  
_______________________________
def alphabet_position(s):
  return " ".join(str(ord(c)-ord("a")+1) for c in s.lower() if c.isalpha())

_______________________________
alphabet = 'abcdefghijklmnopqrstuvwxyz'

def alphabet_position(text):
    if type(text) == str:
        text = text.lower()
        result = ''
        for letter in text:
            if letter.isalpha() == True:
                result = result + ' ' + str(alphabet.index(letter) + 1)
        return result.lstrip(' ')
      
_______________________________
from string import ascii_lowercase
def alphabet_position(text):
    return ' '.join(str(ascii_lowercase.index(n.lower()) + 1) for n in text if n.isalpha())
  
_______________________________
import string

def alphabet_position(text):
    return " ".join([str(string.lowercase.index(letter.lower())+1) for letter in list(text) if letter.isalpha()])
  
_______________________________
def get_positions(text):
    for char in text:
        pos = ord(char)
        if pos >= 65 and pos <= 90:
            yield pos - 64
        if pos >= 97 and pos <= 122:
            yield pos - 96

def alphabet_position(text):
    return " ".join((str(char) for char in get_positions(text)))
  
_______________________________
def alphabet_position(text):
    alphabet = {  'a' : 1,
                  'b' : 2,
                  'c' : 3,
                  'd' : 4,
                  'e' : 5,
                  'f' : 6,
                  'g' : 7,
                  'h' : 8,
                  'i' : 9,
                  'j' : 10,
                  'k' : 11,
                  'l' : 12,
                  'm' : 13,
                  'n' : 14,
                  'o' : 15,
                  'p' : 16,
                  'q' : 17,
                  'r' : 18,
                  's' : 19,
                  't' : 20,
                  'u' : 21,
                  'v' : 22,
                  'w' : 23,
                  'x' : 24,
                  'y' : 25,
                  'z' : 26, }
    inds = []
    for x in text.lower():
        if x in alphabet:
            inds.append(alphabet[x])
    return ' '.join(([str(x) for x in inds]))

_______________________________
def alphabet_position(text):
    
    upper_alpha = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    lower_alpha = "abcdefghijklmnopqrstuvwxyz"
    l = [] 
    
    for i in text:
        if i in upper_alpha:
            index = upper_alpha.index(i) + 1
            l.append(str(index))
        elif i in lower_alpha:
            index = lower_alpha.index(i) + 1
            l.append(str(index)) 
    return " " .join(l)
  
_______________________________
def alphabet_position(text):
  al = 'abcdefghijklmnopqrstuvwxyz'
  return " ".join(filter(lambda a: a != '0', [str(al.find(c) + 1) for c in text.lower()]))
